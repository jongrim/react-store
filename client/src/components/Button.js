import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const style = {
    color: props.color,
    border: 'none',
    backgroundColor: props.bgColor,
    height: props.small ? '20px' : '50px',
    width: props.small ? '80px' : '100px',
    fontSize: props.small ? '12px' : '16px',
    boxShadow: props.boxShadow,
    cursor: 'pointer',
    textAlign: props.textAlign,
    padding: '0px'
  };
  return (
    <button
      className="button"
      style={style}
      onClick={() => {
        props.clickAction(props.dataObj);
      }}>
      {props.children}
    </button>
  );
};

Button.defaultProps = {
  color: 'white',
  bgColor: '#2D5199',
  small: true,
  boxShadow: '0px 0px 6px 2px rgba(0,0,0,0.1)',
  textAlign: 'center'
};

Button.propTypes = {
  color: PropTypes.string,
  bgColor: PropTypes.string,
  small: PropTypes.bool,
  boxShadow: PropTypes.string,
  textAlign: PropTypes.string,
  clickAction: PropTypes.func.isRequired,
  dataObj: PropTypes.object
};
export default Button;
