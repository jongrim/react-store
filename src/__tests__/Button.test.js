import React from 'react';
import Button from '../components/Buttons.js';
import renderer from 'react-test-renderer';

it('Renders a button with text', () => {
  const component = renderer.create(<Button>TEXT</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
