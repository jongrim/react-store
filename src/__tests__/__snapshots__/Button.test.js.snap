// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a button with text 1`] = `
<button
  onClick={[Function]}
  style={undefined}
>
  TEXT
</button>
`;
